src/backend/joanie/core/serializers.py

```diff
@@ -33,23 +198,69 @@ class Meta:
             "products",
         ]
 
+    def to_representation(self, instance):
+        """
+        When user is not authenticated, we return the cached representation,
+        otherwise we generate representation as well
+        """
+        code = self.context.get("code")
+        user = self.context.get("user", None)
+        current_language = get_language()
+
+        if user is None:
+            cache_key = f"course-serializer-{code}-{current_language}-representation"
```

Comment:
Should you already add cache invalidation where necessary? The key computation will have to be shared with invalidation code. Maybe get it via a method on the course instance?

Reply:
Nope I don't do this yet. Good idea!

Reply:
Nope bad idea, as these signals are not send on update

Reply:
we can probably do it explicitly in the order confirmation method?

Reply:
We have to invalidate cache when a course or its related products has been updated not when an order is created.

Reply:
Anyway I'm afraid you won't find anything that triggers even on update... ?

Reply:
Hmm currently, course, products, etc... are updated through django admin panel which seems always perform save. In a second time, we could have an admin api and it will be easy to invalidate cache on put, patch, delete request.

Reply:
Ok but not 24h then :sweat_smile:

Reply:
Yep let's say 1 hour.